engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java

```diff
@@ -81,19 +80,18 @@
  *
  * @author Gunnar Morling
  * @author Hardy Ferentschik
+ * @author Guillaume Smet
  */
 public class AnnotationMetaDataProvider implements MetaDataProvider {
 	private static final Log log = LoggerFactory.make();
-	/**
-	 * The default initial capacity for this cache.
-	 */
-	static final int DEFAULT_INITIAL_CAPACITY = 16;
 
-	protected final ConstraintHelper constraintHelper;
-	protected final TypeResolutionHelper typeResolutionHelper;
-	protected final ConcurrentReferenceHashMap<Class<?>, BeanConfiguration<?>> configuredBeans;
-	protected final AnnotationProcessingOptions annotationProcessingOptions;
-	protected final ValueExtractorManager valueExtractorManager;
+	private final ConstraintHelper constraintHelper;
+	private final TypeResolutionHelper typeResolutionHelper;
+	private final AnnotationProcessingOptions annotationProcessingOptions;
+	private final ValueExtractorManager valueExtractorManager;
+
+	private volatile SoftReference<BeanConfiguration<Object>> objectBeanConfigurationReference;
+	private final Object objectBeanConfigurationReferenceLock = new Object();
```

Comment:
you can avoid creating this object by using synchronized on this AnnotationMetaDataProvider ?